Added Modifiers

BChainbuddy opened the pull request:
# CHANGES
I added a new folder "Modifiers" and added the first smart contract "Owner.sol".
## Owner.sol
onlyOwner is the most used modifier and I think It should definitely be included in the list.
The onlyOwner modifier:
```solidity
    modifier onlyOwner() {
        require(msg.sender == s_owner, "Not the contract owner!!");
        _;
    }
```
The onlyOwner modifier gives the function withdraw restriction that only owner can call it:
```solidity
    function withdraw() public onlyOwner {
        (bool success,)=payable(msg.sender).call{value:address(this).balance}("");
        require(success, "Transfer failed!");
    }
```
The contract also includes a payable function so a user can play with withdraw function in IDE:
```solidity
    function fundContract() public payable {
    }
```

BChainbuddy committed:
Added Modifiers

jinilshah21 commented:
@BChainbuddy You can update the withdraw function. What will happen if the balance is not zero? Add a few lines there. Give the withdraw function some functionality. Also in the fundContract function add the logic by transferring eth from one account to another.

BChainbuddy committed:
updated modifiers

BChainbuddy commented:
@jinilshah21 I have followed your guidelines and updated the code. I have removed the modifiers folder, renamed the file to modifiers.sol and moved it to Basic Functional Contracts for cleaner look. I have updated the parts of the code you pointed out and added new functionalities.
Updated withdraw function:
- [x] Added the require 
- [x] Gave it more functionality by letting the owner decide how much should be withdrawn 
```solidity
    function withdraw(uint256 amount) public onlyOwner {
        require(getContractBalance() >= amount, "There is nothing to withdraw");
        (bool success, ) = payable(msg.sender).call{value: amount}("");
        require(success, "Transfer failed!");
    }
```
I added more functionalities to fundContract function
- [x] Mapping to show how much each address has funded the contract
- [x] Added a new feature that locks the funding function after one year period of funding has finished
- [x] Added contract Balance function
```solidity
    function fundContract() public payable {
        require(isClosed() == false, "The Funding has been closed");
        Funders[msg.sender] += msg.value;
    }
``` 
```solidity
    mapping(address => uint256) public Funders;
    function getFunderAmount(address _address) public view returns (uint256) {
        require(Funders[_address] > 0, "Address not a funder");
        return Funders[_address];
    }
```
```solidity
    uint256 public startedAt;
    function isClosed() public view returns (bool) {
        if (startedAt + 365 days < block.timestamp) {
            return true;
        } else {
            return false;
        }
    }
    function getStartTime() public view returns (uint256) {
        return startedAt;
    }
```
```solidity
    function getContractBalance() public view onlyOwner returns (uint256) {
        return address(this).balance;
    }
```

jinilshah21 reviewed (approved):
Yes the contract looks much better now.
